server/filestore.go

```diff
@@ -639,6 +639,26 @@ func (fs *fileStore) StoreMsg(subj string, hdr, msg []byte) (uint64, int64, erro
 	return seq, ts, nil
 }
 
+// SkipMsg will use the next sequence number but not store anything.
+func (fs *fileStore) SkipMsg() uint64 {
+	// Grab time.
+	now := time.Now().UTC()
+	fs.mu.Lock()
+	seq := fs.state.LastSeq + 1
+	fs.state.LastSeq = seq
+	fs.state.LastTime = now
+	if fs.state.Msgs == 0 {
+		fs.state.FirstSeq = seq
+		fs.state.FirstTime = now
+	}
+	if seq == fs.state.FirstSeq {
```

Comment:
I fail to see when this condition can be true with state.Msgs != 0? Could you describe in which situation would we have one of these 2 "if"'s true, but not the other? (disclaimer, I am not versed on the internal of JS file store implementation).

Reply:
0 is not legal and denotes the start, so that case is special opt. When running, if you have been using a stream for awhile but all messages are now consumed, expired etc. the first seq will be last seq + 1, meaning you are caught up. So if I send 10 messages and consume them, the state of the stream is first = 11, last = 10.